examples/mnist/run.py

```diff
@@ -57,51 +64,43 @@ def normalize(image, label):
   def _build_training_graph(self, training_data):
     """Build a graph for plaintext model training."""
 
-    # model parameters and initial values
-    w0 = tf.Variable(tf.random_normal([28 * 28, 512]))
-    b0 = tf.Variable(tf.zeros([512]))
-    w1 = tf.Variable(tf.random_normal([512, 10]))
-    b1 = tf.Variable(tf.zeros([10]))
-
+    model = keras.Sequential()
+    model.add(keras.layers.Flatten())
+    model.add(keras.layers.Dense(512, input_shape=[28, 28, 1]))
+    model.add(keras.layers.Activation('relu'))
+    model.add(keras.layers.Dense(self.NUM_CLASSES, activation=None))
+    
     # optimizer and data pipeline
     optimizer = tf.train.AdamOptimizer(learning_rate=0.01)
+    def loss(model, x, y):
+      y_ = model(x)
+      return tf.losses.sparse_softmax_cross_entropy(labels=y, logits=y_)
 
-    # training loop
-    def loop_body(i):
+    def grad(model, inputs, targets):
+      with tf.GradientTape() as tape:
+        loss_value = loss(model, inputs, targets)
+      return loss_value, tape.gradient(loss_value, model.trainable_variables)
 
-      # get next batch
+    global_step = tf.Variable(0)
+    def loop_body(i):
       x, y = training_data.get_next()
-
-      # model construction
-      layer0 = tf.matmul(x, w0) + b0
-      layer1 = tf.nn.sigmoid(layer0)
-      layer2 = tf.matmul(layer1, w1) + b1
-
-      predictions = layer2
-      loss = tf.reduce_mean(
-          tf.losses.sparse_softmax_cross_entropy(logits=predictions, labels=y))
-      with tf.control_dependencies([optimizer.minimize(loss)]):
-        return i + 1
-
-    loop = tf.while_loop(lambda i: i < self.ITERATIONS
+      loss_value, grads = grad(model, x, y)
+      update_op = optimizer.apply_gradients(zip(grads, model.trainable_variables), global_step)
+      with tf.control_dependencies([update_op]):
+          return global_step
+    
+    tf.while_loop(lambda i: i < self.ITERATIONS
                          * self.EPOCHS, loop_body, (0,))
-
-    # return model parameters after training
-    with tf.control_dependencies([loop]):
-      print_op = tf.print("Training complete")
-    with tf.control_dependencies([print_op]):
-      return [w0.read_value(),
-              b0.read_value(),
-              w1.read_value(),
-              b1.read_value()]
+    
+    return model.get_weights()
```

Comment:
this should be returning `model.trainable_variables` or `model.trainable_weights`, since `model.get_weights()` returns numpy arrays, which we don't want

Reply:
I've been encountering this error: ```Traceback (most recent call last):
  File "run.py", line 169, in <module>
    params = [tfe.define_private_variable(x) for x in model_owner.provide_input()]  # pylint: disable=E0632
  File "run.py", line 169, in <listcomp>
    params = [tfe.define_private_variable(x) for x in model_owner.provide_input()]  # pylint: disable=E0632
  File "/Users/vinithms/Desktop/Dropout Labs/tf-encrypted/tf_encrypted/protocol/pond/pond.py", line 334, in define_private_variable
    assert isinstance(initial_value, init_val_types), type(initial_value)
AssertionError: <class 'tensorflow.python.ops.resource_variable_ops.ResourceVariable'>```